assignment-client/src/messages/MessagesMixer.cpp

```diff
@@ -88,7 +101,49 @@ void MessagesMixer::sendStatsPacket() {
 }
 
 void MessagesMixer::run() {
+    // wait until we have the domain-server settings, otherwise we bail
+    DomainHandler& domainHandler = DependencyManager::get<NodeList>()->getDomainHandler();
+    connect(&domainHandler, &DomainHandler::settingsReceived, this, &MessagesMixer::domainSettingsRequestComplete);
+
     ThreadedAssignment::commonInit(MESSAGES_MIXER_LOGGING_NAME, NodeType::MessagesMixer);
+
+    startMaxMessagesProcessor();
+}
+
+void MessagesMixer::domainSettingsRequestComplete() {
     auto nodeList = DependencyManager::get<NodeList>();
     nodeList->addSetOfNodeTypesToNodeInterestSet({ NodeType::Agent, NodeType::EntityScriptServer });
+
+    DomainHandler& domainHandler = nodeList->getDomainHandler();
+
+    // parse the settings to pull out the values we need
+    parseDomainServerSettings(nodeList->getDomainHandler().getSettingsObject());
+}
+
+void MessagesMixer::parseDomainServerSettings(const QJsonObject& domainSettings) {
+    const QString MESSAGES_MIXER_SETTINGS_KEY = "messages_mixer";
+    QJsonObject messagesMixerGroupObject = domainSettings[MESSAGES_MIXER_SETTINGS_KEY].toObject();
+
+    const QString NODE_MESSAGES_PER_SECOND_KEY = "max_node_messages_per_second";
+    QJsonValue maxMessagesPerSecondValue = messagesMixerGroupObject.value(NODE_MESSAGES_PER_SECOND_KEY);
+    if (!maxMessagesPerSecondValue.isUndefined()) {
+        _maxMessagesPerSecond = maxMessagesPerSecondValue.toInt();
+    } else {
+        _maxMessagesPerSecond = DEFAULT_NODE_MESSAGES_PER_SECOND;
+    }
+}
+
+void MessagesMixer::processMaxMessagesContainer() {
```

Comment:
```suggestion
void MessagesMixer::clearMaxMessagesContainer() {
```
Possibly a better name?

Reply:
I like the name "process" as it implies that other things can also happen during the clear if we wanted to add functionality later.